test/Confluent.Kafka.UnitTests/Consumer.cs

```diff
@@ -33,6 +39,8 @@ public void Constuctor()
             Assert.True(e.Message.Contains("group.id"));
             e = Assert.Throws<ArgumentException>(() => { var c = new Consumer<Null, string>(config, null, new StringDeserializer(Encoding.UTF8)); });
             Assert.True(e.Message.Contains("group.id"));
+
+            // positve case covered by integration tests. here, avoiding creating a rd_kafka_t instance.
```

Comment:
I would argue that this is a typical unit-test and that we shouldn't rely on integration tests in this case - they are not run as frequently, and not from CI.

Apart from that this is LGTM.

Your call on the unit test.